django/librarian/views.py

```diff
@@ -430,7 +430,8 @@ def data_source_stop(request, data_source_id):
     # Stop all celery tasks for documents within this data source
     data_source = get_object_or_404(DataSource, id=data_source_id)
     for document in data_source.documents.all():
-        document.stop()
+        if document.status in ["PENDING", "INIT", "PROCESSING"]:
+            document.stop()
```

Comment:
The "stop" button should really only stop documents that are currently being processed.